build/ci/community/wait_demo_pr_checks.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash -xe
+set +o pipefail
+
+gh_pr_checks_command (){
+  community_operators_branch=$1
+  forked_repository=$2
+  gh pr checks $community_operators_branch --repo $forked_repository
+}
+
+wait_for_checks_to_start(){
+  community_operators_branch=$1
+  forked_repository=$2
+  while [ `gh_pr_checks_command $community_operators_branch $forked_repository | wc -l` -eq 0 ]; do
+    sleep 1
+  done
+}
+wait_for_checks_to_complete(){
+  community_operators_branch=$1
+  forked_repository=$2
+  all_tests_passed=false
+  repo_pr=`gh pr list --repo $forked_repository | grep $community_operators_branch`
+  if [[ "$repo_pr" == *"$community_operators_branch"* ]]; then
```

Comment:
could you give an example for an output of the `gh pr list` command that would result in the `if` condition being false?

Reply:
and what does it mean if there is no pr with the required branch? shouldn't it fail?

Reply:
in addition, I see this code in at least 3 different places in this PR.
please move it to a common location